src/main/java/frc/trigon/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,108 @@
+    private final TalonFXMotor
+            rightMotor = ShooterConstants.RIGHT_MOTOR,
+            leftMotor = ShooterConstants.LEFT_MOTOR;
+    private final VelocityTorqueCurrentFOC velocityRequest = new VelocityTorqueCurrentFOC(0);
+    private final TorqueCurrentFOC torqueRequest = new TorqueCurrentFOC(0);
+    private double
+            targetRightVelocityRotationsPerSecond = 0,
+            targetLeftVelocityRotationsPerSecond = 0;
+
+    public Shooter() {
+        setName("Shooter");
+    }
+
+    @Override
+    public void updateLog(SysIdRoutineLog log) {
+        log.motor("RightShooter")
+                .angularPosition(Units.Rotations.of(rightMotor.getSignal(TalonFXSignal.POSITION)))
+                .angularVelocity(Units.RotationsPerSecond.of(rightMotor.getSignal(TalonFXSignal.VELOCITY)))
+                .voltage(Units.Volts.of(rightMotor.getSignal(TalonFXSignal.TORQUE_CURRENT)));
+        log.motor("LeftShooter")
+                .angularPosition(Units.Rotations.of(leftMotor.getSignal(TalonFXSignal.POSITION)))
+                .angularVelocity(Units.RotationsPerSecond.of(leftMotor.getSignal(TalonFXSignal.VELOCITY)))
+                .voltage(Units.Volts.of(leftMotor.getSignal(TalonFXSignal.TORQUE_CURRENT)));
+    }
+
+    @Override
+    public void stop() {
+        rightMotor.stopMotor();
+        leftMotor.stopMotor();
+        ShooterConstants.RIGHT_MECHANISM.setTargetVelocity(0);
+        ShooterConstants.LEFT_MECHANISM.setTargetVelocity(0);
+    }
+
+    @Override
+    public void periodic() {
+        rightMotor.update();
+        leftMotor.update();
+        updateMechanism();
+    }
+
+    @Override
+    public void drive(Measure<Voltage> voltageMeasure) {
+        rightMotor.setControl(torqueRequest.withOutput(voltageMeasure.in(Units.Volts)));
+        leftMotor.setControl(torqueRequest.withOutput(voltageMeasure.in(Units.Volts)));
+    }
+
+    @Override
+    public SysIdRoutine.Config getSysIdConfig() {
+        return ShooterConstants.SYSID_CONFIG;
+    }
+
+    public boolean bothMotorsAtTargetVelocity() {
+        return atTargetRightVelocity() && atTargetLeftVelocity();
+    }
+
+    public boolean atTargetRightVelocity() {
+        return Math.abs(rightMotor.getSignal(TalonFXSignal.VELOCITY) - targetRightVelocityRotationsPerSecond) < ShooterConstants.VELOCITY_TOLERANCE;
+    }
+
+    public boolean atTargetLeftVelocity() {
+        return Math.abs(leftMotor.getSignal(TalonFXSignal.VELOCITY) - targetLeftVelocityRotationsPerSecond) < ShooterConstants.VELOCITY_TOLERANCE;
+    }
+
+    void reachTargetShootingVelocityFromShootingCalculations() {
+        final double
+                targetRightVelocityRotationsPerSecond = shootingCalculations.getTargetShootingState().targetShootingVelocityRotationsPerSecond(),
+                targetLeftVelocityRotationsPerSecond = targetRightVelocityRotationsPerSecond * ShooterConstants.LEFT_MOTOR_TO_RIGHT_MOTOR_RATIO;
```

Comment:
`ShooterConstants.LEFT_MOTOR_TO_RIGHT_MOTOR_RATIO` is actually used as `RIGHT_MOTOR_TO_LEFT_MOTOR_RATIO`